mantidproject/mantid
# Add Polarization Efficiencies to GUI

# BLOCKED BY: 
- [x] #34805

**Description of work.**
Adds a GUI element to allow selection of a workspace for the new Polarization Efficiencies property of the Refl Load and Process algorithm. 

The logic is set up such that the check box enables and disables the polarization correction, then the combo box allows selection of workspaces or to be left blank should the user wish to use the values in the parameter file. 

**To test:**
1. Talk to Conor or Rachel about getting the corrections file. It's not public, so it can't be put here. 
2. Load the file into mantid
3. Boot up the Refl GUI
4. Switch the instrument to POLREF.
5. Go to the Experiment Settings Tab
6. Enable polarization corrections
7. Check the combo box gets enabled. 
8. Select the downloaded workspace
9. Go back to the Runs tab
10. Enter `31835` into the Run column (not the group, the second box down) and `1.0` into the angle column
11. Click process
12. Check that the `IvsLam_31835` workspace group contains a workspace ending with `_--` and `_++`

Fixes #34744 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
